.github/workflows/static-code-analysis.yml

```diff
@@ -0,0 +1,51 @@
+name: "Static code analysis"
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  schedule:
+    # Everyday at midnight
+    - cron: '0  0 * * *'
+
+jobs:
+  codeql:
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: >
+           sudo apt update &&
+           sudo apt install -y autoconf automake autopoint check cifs-utils
+           clang dh-apparmor dnsutils docbook-xml docbook-xsl gettext
+           krb5-config libc-ares-dev libcmocka-dev libcollection-dev
+           libdbus-1-dev libdhash-dev libglib2.0-dev libini-config-dev
+           libkeyutils-dev libkrb5-dev libldap2-dev libldb-dev libltdl-dev
+           libnfsidmap-dev libnl-3-dev libnl-route-3-dev libpam0g-dev
+           libpcre2-dev libpopt-dev libsasl2-dev libselinux1-dev
+           libsemanage1-dev libsmbclient-dev libsystemd-dev libtalloc-dev
+           libtdb-dev libtevent-dev libtool libtool-bin libxml2-utils make
+           python-dev python3-dev samba-dev systemd xml-core xsltproc
+           libssl-dev fakeroot faketime libnss-wrapper libuid-wrapper
+           libpam-wrapper python-pytest python3-ldap python3-ldb
+           python3-requests python-psutil ldap-utils slapd systemtap-sdt-dev
+           libhttp-parser-dev libcurl4-openssl-dev krb5-kdc
+           krb5-admin-server krb5-user uuid-dev dbus python-dbus pep8
+           libssl-dev gnutls-bin softhsm2 libp11-kit-dev libunistring-dev
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+
+      - name: Compile sssd
+        run: |
+           source contrib/fedora/bashrc_sssd
+           reconfig --with-smb-idmap-interface-version=5
```

Comment:
No, it wasn't detected so I had to force it manually.

Reply:
Can you check if this is also used in official ubuntu package? Or if there is something we can do to improve detection on ubuntu? It would be better to have this automated.

Reply:
Ubuntu is using this option to configure and we've also been [using it for debian](https://github.com/SSSD/sssd/blob/master/contrib/ci/configure.sh#L64) for the last 4 years.

As a side note, I wanted to use `chmake`, but it timed out with no apparent reason. I'll add additional debug logs (newly discovered) but I don't know if I'll be able to fix it.

Reply:
I'm fine with that.

Reply:
Thank you for looking into it.

Reply:
Sorry, my previous comment about `chmake` wasn't correct and I didn't change anything there. Please review the new version of the file and tell me what you think.